pkg/version/update.go

```diff
@@ -63,3 +72,135 @@ func CheckForUpdate(
 
 	return &updateCheck, nil
 }
+
+func LogUpdateResponse(ctx context.Context, ucr *UpdateCheckResponse) {
+	if ucr != nil && ucr.Message != "" {
+		log.Ctx(ctx).Info().Str("NewVersion", ucr.Version.GitVersion).Msg(strings.ReplaceAll(ucr.Message, "\n", " "))
+	}
+}
+
+// RunUpdateChecker starts a goroutine that will periodically make an update
+// check according to the configured update frequency and check skipping. The
+// goroutine is ended by canceling the passed context. `getServerVersion` is
+// allowed to return a nil version if there is no server to communicate with
+// (e.g. because the node running the update check is the server).
+func RunUpdateChecker(
+	ctx context.Context,
+	getServerVersion func(context.Context) (*models.BuildVersionInfo, error),
+	responseCallback func(context.Context, *UpdateCheckResponse),
+) {
+	updateFrequency := config.GetUpdateCheckFrequency()
+	if updateFrequency <= 0 {
+		log.Ctx(ctx).Warn().Dur(types.UpdateCheckFrequency, updateFrequency).Msg("Update frequency is zero or less so no update checks will run")
+		return
+	}
+
+	clientVersion := Get()
+	clientID, err := config.GetClientID()
+	if err != nil {
+		log.Ctx(ctx).Error().Err(err).Msg("Failed to read client ID")
+		return
+	}
+
+	runUpdateCheck := func() {
+		// Check this every time, to handle programmatic changes to config that
+		// may switch this on or off.
+		if skip, err := config.Get[bool](types.UpdateSkipChecks); skip || err != nil {
+			log.Ctx(ctx).Debug().Err(err).Bool(types.UpdateSkipChecks, skip).Msg("Skipping update check due to config")
+			return
+		}
+
+		// The server may update itself between checks, so always ask the server
+		// for its current version.
+		serverVersion, err := getServerVersion(ctx)
+		if err != nil {
+			log.Ctx(ctx).Error().Err(err).Msg("Failed to read server version")
+			serverVersion = nil
+		}
+
+		updateResponse, err := CheckForUpdate(ctx, clientVersion, serverVersion, clientID)
+		if err != nil {
+			log.Ctx(ctx).Error().Err(err).Msg("Failed to perform update check")
+		}
+
+		if err == nil {
+			responseCallback(ctx, updateResponse)
+			err = writeNewLastCheckTime()
+			log.Ctx(ctx).WithLevel(logger.ErrOrDebug(err)).Err(err).Msg("Completed update check")
+		}
+	}
+
+	lastCheck, err := readLastCheckTime()
+	if err != nil {
+		log.Ctx(ctx).Warn().Err(err).Msg("Error reading update check state – will perform check anyway")
+		lastCheck = time.UnixMilli(0)
+	}
+
+	// Count down the remaining time between the last check and the next check,
+	// and then reset the ticker to start doing regular periodic checks.
```

Comment:
```suggestion
	// Count down the remaining time between the last check and the next check,
	// and then reset the ticker to start doing regular periodic checks. This is fine 
	// because the ticker will not have fired before the initial timer.
```

super-minor suggestion to make it a tiny bit clearer